# Создайте собственный класс-исключение, обрабатывающий ситуацию деления на нуль.
# Проверьте его работу на данных, вводимых пользователем.
# При вводе пользователем нуля в качестве делителя программа должна корректно обработать эту ситуацию и не завершиться с ошибкой.


class OwnError(Exception):
    msg = 'Ноль не может быть делителем!'


try:
    x = int(input("Введите Делимое: "))
    y = int(input("Введите Делитель: "))
    if y == 0:
        raise OwnError
except OwnError as e:
    print(e.msg)
except:
    print("Что-то пошло не так")
else:
    print(f"Все хорошо. Результат деления: {x / y}")
